Sources/JWTKit/ECDSA/ECDSAKey.swift

```diff
@@ -1,192 +1,27 @@
-        }
-        defer { CJWTKitBoringSSL_X509_free(x509) }
-        let pkey = CJWTKitBoringSSL_X509_get_pubkey(x509)
-        defer { CJWTKitBoringSSL_EVP_PKEY_free(pkey) }
-
-        guard let c = CJWTKitBoringSSL_EVP_PKEY_get1_EC_KEY(pkey) else {
-            throw JWTError.signingAlgorithmFailure(ECDSAError.newKeyByCurveFailure)
-        }
-        return self.init(c)
-    }
-    
-    public static func `public`(pem string: String) throws -> ECDSAKey {
-        try .public(pem: [UInt8](string.utf8))
-    }
-
-    public static func `public`<Data>(pem data: Data) throws -> ECDSAKey
-        where Data: DataProtocol
-    {
-        let c = try self.load(pem: data) { bio in
-            CJWTKitBoringSSL_PEM_read_bio_EC_PUBKEY(bio, nil, nil, nil)
-        }
-        return self.init(c)
-    }
-
-    public static func `private`(pem string: String) throws -> ECDSAKey {
-        try .private(pem: [UInt8](string.utf8))
-    }
-
-    public static func `private`<Data>(pem data: Data) throws -> ECDSAKey
-        where Data: DataProtocol
-    {
-        let c = try self.load(pem: data) { bio in
-            CJWTKitBoringSSL_PEM_read_bio_ECPrivateKey(bio, nil, nil, nil)
-        }
-        return self.init(c)
-    }
-
-    let c: OpaquePointer
-
-    init(_ c: OpaquePointer) {
-        self.c = c
-    }
-    
-    public convenience init(parameters: Parameters, curve: Curve = .p521, privateKey: String? = nil) throws {
-        guard let c = CJWTKitBoringSSL_EC_KEY_new_by_curve_name(curve.cName) else {
-            throw JWTError.signingAlgorithmFailure(ECDSAError.newKeyByCurveFailure)
-        }
-        
-        guard let bnX = BigNumber(base64URL: parameters.x) else {
-            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to interpret x as BN")
-        }
-        guard let bnY = BigNumber(base64URL: parameters.y) else {
-            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to interpret y as BN")
-        }
-
-        if CJWTKitBoringSSL_EC_KEY_set_public_key_affine_coordinates(c, bnX.c, bnY.c) != 1 {
-            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to set public key")
-        }
-        
-        if let privateKey = privateKey {
-            guard let bnPrivate = BigNumber(base64URL: privateKey) else {
-                throw JWTError.generic(identifier: "ecPrivateKey", reason: "Unable to interpret privateKey as BN")
-            }
-            if CJWTKitBoringSSL_EC_KEY_set_private_key(c, bnPrivate.c) != 1 {
-                throw JWTError.generic(identifier: "ecPrivateKey", reason: "Unable to set private key")
-            }
-        }
-
-        self.init(c)
-    }
+public protocol PrivateKey {
+    func signature<D>(for data: D) throws -> Foundation.Data where D: Foundation.DataProtocol
+}
+public protocol PublicKey {
+    func isValidSignature<D>(_ signature: Foundation.Data, for data: D) throws -> Bool where D: Foundation.DataProtocol
+}
```

Comment:
These names are too nonspecific to be `public`. Why does the `ECDSAKey` need to be a protocol in the first place? And I hasten to point out also that removing all of this public API will cause a massive semver breakage.